apis/cloudidentity/v1beta1/membership_types.go

```diff
@@ -0,0 +1,178 @@
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+var CloudIdentityMembershipGVK = GroupVersion.WithKind("CloudIdentityMembership")
+
+// CloudIdentityMembershipSpec defines the desired state of CloudIdentityMembership
+// +kcc:proto=mockgcp.cloud.cloudidentity.groups.v1beta1.Membership
+type CloudIdentityMembershipSpec struct {
+	/* Immutable. Optional. The service-generated name of the resource. Used for acquisition only. Leave unset to create a new resource. */
+	ResourceID *string `json:"resourceID,omitempty"`
+
+	// Immutable.
+	// +required
+	GroupRef refsv1beta1.GroupRef `json:"groupRef"`
+
+	// Immutable. The `EntityKey` of the member. Either `member_key` or `preferred_member_key` must be set when calling MembershipsService.CreateMembership but not both; both shall be set when returned.
+	MemberKey *MembershipMemberKey `json:"memberKey,omitempty"`
+
+	// Immutable. Required. Immutable. The `EntityKey` of the member.
+	// +required
+	PreferredMemberKey MembershipPreferredMemberKey `json:"preferredMemberKey"`
+
+	// The `MembershipRole`s that apply to the `Membership`. If unspecified, defaults to a single `MembershipRole` with `name` `MEMBER`. Must not contain duplicate `MembershipRole`s with the same `name`.
+	// +required
+	Roles []MembershipRoles `json:"roles"`
+}
+
+// CloudIdentityMembershipStatus defines the config connector machine state of CloudIdentityMembership
+type CloudIdentityMembershipStatus struct {
+	commonv1alpha1.CommonStatus `json:",inline"`
+
+	// ObservedState is the state of the resource as most recently observed in GCP.
+	ObservedState *CloudIdentityMembershipObservedState `json:"observedState,omitempty"`
+
+	// Output only. The time when the `Membership` was created.
+	CreateTime *string `json:"createTime,omitempty"`
+
+	/* Output only. Delivery setting associated with the membership. Possible values: DELIVERY_SETTING_UNSPECIFIED, ALL_MAIL, DIGEST, DAILY, NONE, DISABLED */
+	DeliverySetting *string `json:"deliverySetting,omitempty"`
+
+	// displayName: This field does not exist in the v1/v1beta1/v1alpha1 proto defn.
+
+	// Output only. The display name of this member, if available
+	DisplayName *MembershipDisplayNameStatus `json:"displayName,omitempty"`
+
+	/* Output only. The type of the membership. Possible values: OWNER_TYPE_UNSPECIFIED, OWNER_TYPE_CUSTOMER, OWNER_TYPE_PARTNER */
+	Type *string `json:"type,omitempty"`
+
+	// Output only. The time when the `Membership` was last updated.
+	UpdateTime *string `json:"updateTime,omitempty"`
+}
+
+// CloudIdentityMembershipObservedState is the state of the CloudIdentityMembership resource as most recently observed in GCP.
+// +kcc:proto=mockgcp.cloud.cloudidentity.groups.v1beta1.Membership
+type CloudIdentityMembershipObservedState struct {
+	// The name of the Membership resource in GCP. Server generated.
+	// Shall be of the form `groups/{group_id}/memberships/{membership_id}`.
+	Name string `json:"name,omitempty"`
```

Comment:
Is this field equivalent to `externalRef`? If so, let's remove it. It is probably not clearly documented in the guide, but the latest chat room discussion seems to be leaning in this direction. And if we removed the "name" field, we should just remove the "ObservedState" and place the `// +kcc:proto=mockgcp.cloud.cloudidentity.groups.v1beta1.Membership` annotation to the "Status", it will help the mapper generation.

Reply:
Its the server generated ID and is called name in the proto. Should'nt we have something similar here ?

Reply:
removed.